evalml/tests/automl_tests/dask_tests/test_automl_dask.py

```diff
@@ -27,10 +31,65 @@ def cluster():
     dask_cluster.close()
 
 
-def test_automl(X_y_binary_cls, cluster, sequential_engine):
+@pytest.fixture(scope="module")
+def thread_pool():
+    pool = ThreadPoolExecutor()
+    yield pool
+    pool.shutdown()
+
+
+@pytest.fixture(scope="module")
+def process_pool():
+    pool = ProcessPoolExecutor()
+    yield pool
+    pool.shutdown()
+
+
+# List of tuples indicating which engine types and resource pools to parametrize tests over.
+# Removed '("CFEngine", "process")' as it doesn't work properly on GitHub test runners.
+engine_and_resource_types = [("CFEngine", "thread"), ("DaskEngine", "N/A")]
+
+
+def _get_engine_support(parallel_engine_type, thread_pool, cluster):
+    """Helper function to return the proper combination of resource pool, client class and
+    engine class for testing purposes.
+
+    e.g. The CFEngine can be run either with a ThreadPoolExecutor or a ProcessPoolExecutor,
+    so _get_engine_support("CFEngine", thread_pool, process_pool, "thread", cluster) returns a
```

Comment:
Also, I feel like there's some potential for refactoring that I can't quite put my finger on: if all we're checking for is `parallel_engine_type`, is it possible to do:

```
def _get_engine_support(parallel_engine_type, resource):
    if parallel_engine_type == "CFEngine":
        client_class = CFClient
        engine_class = CFEngine
        # thread_pool should be passed as resource
    elif parallel_engine_type == "DaskEngine":
        client_class = dd.Client
        engine_class = DaskEngine
        # cluster should be passed as resource
    return resources, client_class, engine_class
```

Reply:
Sorry, this is a weird helper function.  It's gone through a few iterations and ultimately looks like a lot of hullabaloo for little return.  I think the reason I didn't do it the way you mentioned was to leave the decision logic for what resources should be entirely inside the function rather than have an implied knowledge in the hosting function/test.

Reply:
@angela97lin I think there's good potential for refactoring in the engine code.  I filed [this](https://github.com/alteryx/evalml/issues/2534) to cover that.